### PR TITLE
flexi: new package

### PR DESCRIPTION
**Pull request by michaelkuhn**

### CODE REVIEW

File: var/spack/repos/builtin/packages/flexi/package.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                if flags is None:
+                    flags = []
+                flags.append('-fallow-argument-mismatch')
+
+        return (flags, None, None)
+
+    def cmake_args(self):
+        args = [
+            '-DFLEXI_BUILD_HDF5=OFF'
+        ]
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/flexi/package.py]: We can:
```suggestion
        args = [
            '-DFLEXI_BUILD_HDF5=OFF',
            self.define_from_variant('FLEXI_MPI', 'mpi')
        ]
```
and delete the lines below.

File: var/spack/repos/builtin/packages/flexi/package.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        if '+mpi' in self.spec:
+            args.append('-DFLEXI_MPI=ON')
+        else:
+            args.append('-DFLEXI_MPI=OFF')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/flexi/package.py]:
```suggestion
```

**Review by alalazo — COMMENTED** [review]
I think we can simplify a bit the recipe.

**Comment by michaelkuhn** [comment]
I always forget about those, thanks! I have pushed an updated version (the patch is also not necessary anymore).

**Review by alalazo — APPROVED** [review]

**Pull request merged**